tests/system/Database/Live/ForgeTest.php

```diff
@@ -1475,4 +1475,219 @@ public function testDropPrimaryKey()
+
+            foreach ($dummyData as $singleDummyData) {
+                $this->db->table($table)->insert($singleDummyData);
+            }
+        }
+
+        $fields = [
+            'userid'      => ['type' => 'int', 'constraint' => 9],
+            'userid2'     => ['type' => 'int', 'constraint' => 9],
+            'category'    => ['type' => 'varchar', 'constraint' => 63],
+            'name'        => ['type' => 'varchar', 'constraint' => 63],
+            'uid'         => ['type' => 'varchar', 'constraint' => 63],
+            'class'       => ['type' => 'varchar', 'constraint' => 63, 'null' => true],
+            'input'       => ['type' => 'varchar', 'constraint' => 63, 'null' => true],
+            'role'        => ['type' => 'varchar', 'constraint' => 63, 'default' => ''],
+            'icon'        => ['type' => 'varchar', 'constraint' => 63, 'default' => ''],
+            'summary'     => ['type' => 'varchar', 'constraint' => 255, 'default' => ''],
+            'description' => ['type' => 'text', 'null' => true],
+            'created_at'  => ['type' => 'datetime', 'null' => true],
+            'updated_at'  => ['type' => 'datetime', 'null' => true],
+            'deleted_at'  => ['type' => 'datetime', 'null' => true],
+        ];
+
+        $this->forge->addField('id');
+        $this->forge->addField($fields);
+
+        $this->forge->addKey('name', false, false, 'db_actions_name');
+        $this->forge->addKey('uid', false, false, 'db_actions_uid');
+        $this->forge->addKey(['category', 'name'], false, false, 'db_actions_category_name');
+        $this->forge->addKey(['deleted_at', 'id'], false, false, 'db_actions_deleted_at');
+        $this->forge->addKey('created_at', false, false, 'db_actions_created_at');
+
+        if ($this->db->DBDriver === 'SQLite3') {
+            $this->forge->addForeignKey('userid', 'user', 'id');
+            $this->forge->addForeignKey('userid2', 'user2', 'id');
+        } else {
+            $this->forge->addForeignKey('userid', 'user', 'id', '', '', 'db_actions_userid_foreign');
+            $this->forge->addForeignKey('userid2', 'user2', 'id', '', '', 'db_actions_userid2_foreign');
+        }
+
+        $this->forge->createTable('actions');
+
+        // now drop columns and indexes
+        $this->forge->dropKey('actions', 'db_actions_category_name', false);
+
+        $indexes = array_filter(
+            $this->db->getIndexData('actions'),
+            static fn ($index) => ($index->name === 'db_actions_category_name')
+                    && ($index->fields === [0 => 'category', 1 => 'name'])
+        );
+        $this->assertCount(0, $indexes);
+
+        $this->forge->dropForeignKey('actions', 'db_actions_userid_foreign');
+        $this->forge->dropForeignKey('actions', 'db_actions_userid2_foreign');
+
+        $this->assertCount(0, $this->db->getForeignKeyData('actions'));
+
+        // redefine id without auto increment
+        if ($this->db->DBDriver === 'MySQLi') {
```

Comment:
Why only MySQLi needs this?

Reply:
I removed this. But the reason is that you can't drop a primary key in MySQL if it is auto increment. You will get an incorrect table definition error. MySQL requires an auto increment column to be defined as a key.. so if you drop the key then the rule is violated.